src/hooks/useHandleRequestError.js

```diff
@@ -0,0 +1,31 @@
+import { useIntl } from 'react-intl';
+
+function getHoustonErrorMessage(error) {
+  return error?.response?.data?.message;
+}
+
+function getAxiosErrorMessage(error) {
+  return error?.toJSON ? error.toJSON().message : null;
+}
+
+function getJsErrorMessage(error) {
+  return error?.message;
+}
+
+export default function useHandleRequestError() {
+  const intl = useIntl();
+
+  return function handleRequestError(error) {
+    const defaultErrorMessage = intl.formatMessage({
+      id: 'UNKNOWN_ERROR',
+    });
+
+    const errorMessage =
+      getHoustonErrorMessage(error) ||
+      getAxiosErrorMessage(error) ||
+      getJsErrorMessage(error) ||
+      defaultErrorMessage;
+
+    return Promise.reject(errorMessage);
+  };
+}
```

Comment:
This is largely a copy of `formatError` in `useMutate`
https://github.com/WildMeOrg/codex-frontend/blob/d558ff2fd6fed3b5b9980b8080814ce586562b73/src/hooks/useMutate.js#L12